pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -200,11 +308,10 @@ func (e *DockercfgController) SetDockerURLs(newDockerURLs ...string) {
 }
 
 func needsDockercfgSecret(serviceAccount *api.ServiceAccount) bool {
-
 	mountableDockercfgSecrets, imageDockercfgPullSecrets := getGeneratedDockercfgSecretNames(serviceAccount)
 
 	// look for an ImagePullSecret in the form
-	if len(imageDockercfgPullSecrets) > 0 && len(mountableDockercfgSecrets) > 0 {
+	if len(imageDockercfgPullSecrets) > 0 || len(mountableDockercfgSecrets) > 0 {
```

Comment:
revert this... `needsDockercfgSecret` means we have to do work on the service account, even if that work is just copying an existing dockercfg from one list to the other